### PR TITLE
Support Google Storage for artifact logging 

### PR DESCRIPTION
**Pull request by dsuhinin**
What's inside the box:
- new `gs` adapter has been added to support Google Storage for artifact logging.

### CODE REVIEW

File: pkg/api/mlflow/service/artifact/storage/storage.go
Comment (jgiannuzzi) [inline pkg/api/mlflow/service/artifact/storage/storage.go:1]: General remark about this factory now that we're adding a 3rd storage: I _think_ we should simplify it, as there is a lot of duplicated code. We could do this:
* try to load and return
* if not found, look at the name (switch/case) and create accordingly
* store and return
I don't think there are real benefits in checking whether we got the right kind of storage from the map — something would be incredibly wrong in our code for this to happen, and simplifying it as suggested above should help prevent that

File: pkg/api/mlflow/service/artifact/storage/gs.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	}
+	return &GS{
+		client: client,
+	}, nil
+}
+
+// List implements ArtifactStorageProvider interface.
+func (s GS) List(ctx context.Context, artifactURI, path string) ([]ArtifactObject, error) {
+	// 1. process input parameters.
+	bucket, rootPrefix, err := ExtractS3BucketAndPrefix(artifactURI)
```
Comment (jgiannuzzi) [inline pkg/api/mlflow/service/artifact/storage/gs.go:41]: we should rename this method

Comment (dsuhinin) [reply pkg/api/mlflow/service/artifact/storage/gs.go]: renamed it to be a bit more generic.

File: pkg/api/mlflow/service/artifact/storage/helpers.go [hunk 2]
```diff
@@ -7,11 +7,12 @@ import (
 	"github.com/rotisserie/eris"
 )
 
-// ExtractS3BucketAndPrefix extracts bucket name and prefix path from provided uri.
-// after processing of this uri s3://fasttrackml/2/30357ed2eaac4f2cacdbcd0e06e9e48a/artifacts result will be:
+// ExtractBucketAndPrefix extracts bucket name and prefix path from provided uri.
+// uri could be any S3 compatible storage uri like; AWS S3 or Google Storage.
```
Comment (jgiannuzzi) [inline pkg/api/mlflow/service/artifact/storage/helpers.go:11]: tiny typo:
```suggestion
// uri could be any S3 compatible storage uri like AWS S3 or Google Storage.
```

File: pkg/api/mlflow/service/artifact/storage/storage.go
Comment (dsuhinin) [reply pkg/api/mlflow/service/artifact/storage/storage.go:1]: addressed.

File: pkg/api/mlflow/service/artifact/storage/storage.go [hunk 3]
```diff
@@ -71,47 +71,34 @@ func (s *ArtifactStorageFactory) GetStorage(
...
 		if err != nil {
 			return nil, eris.Wrap(err, "error initializing local artifact storage")
 		}
-		s.storageList.Store(LocalStorageName, storage)
-		return storage, nil
 	default:
 		return nil, eris.Errorf("unsupported schema has been provided: %s", u.Scheme)
 	}
+
+	s.storageList.Store(S3StorageName, storage)
```
Comment (jgiannuzzi) [inline pkg/api/mlflow/service/artifact/storage/storage.go:102]: oops! ❌ your're registering everything as S3!!!!

Comment (dsuhinin) [reply pkg/api/mlflow/service/artifact/storage/storage.go]: yep, yep. fixed. my bad.

File: pkg/api/mlflow/service/artifact/storage/helpers.go (hunk 2 above)
Comment (dsuhinin) [reply pkg/api/mlflow/service/artifact/storage/helpers.go]: done.

**Review by jgiannuzzi — DISMISSED** [review]
LGTM 👍
I also tested it locally and can confirm it works as expected.

**Comment by jgiannuzzi** [comment]
@dsuhinin could you please also add `google-cloud-storage` to the `Pipfile`?
```python
pipenv install --dev google-cloud-storage
```

**Comment by dsuhinin** [comment]
> @dsuhinin could you please also add `google-cloud-storage` to the `Pipfile`?
> 
> ```python
> pipenv install --dev google-cloud-storage
> ```

yep. done!

**Comment by jgiannuzzi** [comment]
it looks like the `Pipfile.lock` did not get updated!

**Comment by dsuhinin** [comment]
> it looks like the `Pipfile.lock` did not get updated!

yep, yep. didn't commit it by some reason. now should be good.

**Review by jgiannuzzi — APPROVED** [review]

**Pull request merged**